x/crosschain/keeper/abci.go

```diff
@@ -175,3 +204,81 @@ func CheckAndUpdateCctxGasPrice(
 
 	return gasPriceIncrease, additionalFees, nil
 }
+
+// CheckAndUpdateCCTXGasPriceBTC updates the gas price for the given Bitcoin chain CCTX
+func CheckAndUpdateCCTXGasPriceBTC(
+	ctx sdk.Context,
+	k Keeper,
+	medianGasPrice math.Uint,
+	cctx types.CrossChainTx,
+	flags observertypes.GasPriceIncreaseFlags,
+) (gasPriceIncrease math.Uint, additionalFees math.Uint, err error) {
+	// check retry interval -- use default interval if not set
+	retryIntervalBTC := flags.RetryIntervalBTC
+	if retryIntervalBTC <= 0 {
+		retryIntervalBTC = observertypes.DefaultGasPriceIncreaseFlags.RetryIntervalBTC
+	}
+
+	// skip if Bitcoin gas price retry interval is not reached
+	lastUpdated := time.Unix(cctx.CctxStatus.LastUpdateTimestamp, 0)
+	if ctx.BlockTime().Before(lastUpdated.Add(retryIntervalBTC)) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// get current gas price
+	currentGasPrice, err := cctx.GetCurrentOutboundParam().GetGasPriceUInt64()
+	if err != nil {
+		return math.ZeroUint(), math.ZeroUint(), err
+	}
+
+	// use latest median gas price as new gas price, the reasons are:
+	// 1. the goal is to increase the average gas price of all the stuck txs to market level
+	// 2. zetaclient can't replace stuck tx individually, it only gives more funds to the last stuck tx (child tx)
+	// 3. updating all pending CCTXs to the same 'mediaGasPrice' number simplyfies the calculation in zetaclient
+	newGasPrice := medianGasPrice
+	if math.NewUint(currentGasPrice).GTE(newGasPrice) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// compute gas price increase
+	gasPriceIncrease = newGasPrice.SubUint64(currentGasPrice)
+
+	// withdraw additional fees from the gas stability pool
+	chainID := cctx.GetCurrentOutboundParam().ReceiverChainId
+	gasLimit := math.NewUint(cctx.GetCurrentOutboundParam().CallOptions.GasLimit)
```

Comment:
We don't use this paramter with Bitcoin, don't we?

Reply:
Yeah. `Zetaclient` is not using it.

It is only used to charge withdrawer fee. It is a static number `gasLimit == 177`, meaning each withdrawal costs `177vB` in Bitcoin transaction size, and we charge withdraw fee as `177vB * gasPrice`. When `gasPrice` increases, it charges extra `177vB * extraGasPrice`